kong/tools/http.lua

```diff
@@ -527,4 +527,35 @@ do
 end
 
 
+do
+  local replace_dashes = require("kong.tools.string").replace_dashes
+
+  function _M.get_header(name, ctx)
+    local value = ngx.var["http_" .. replace_dashes(name)]
+
+    if not value or not value:find(", ", 1, true) then
+      return value
+    end
+
+    local headers
+
+    if ctx then
+      if not ctx.cached_request_headers then
+        ctx.cached_request_headers = ngx.req.get_headers()
+      end
+
+      headers = ctx.cached_request_headers
+
+    else
+      headers = ngx.req.get_headers()
+    end
```

Comment:
Please remove the cachem and `ctx` argument. Let's solve this properly in perf pr after this.

Reply:
Oh, it is already used, but just internally. Thus I am marking this as resolved.

Reply:
done.